deepchem/feat/molecule_featurizers/one_hot_featurizer.py

```diff
@@ -88,14 +96,14 @@ def _featurize_string(self, string: str) -> np.ndarray:
       The shape is `(max_length, len(charset) + 1)`.
       The index of unknown character is `len(charset)`.
     """
-    # validation
-    if (len(string) > self.max_length):
-      logger.info(
-          "The length of {} is longer than `max_length`. So we return an empty array."
-      )
-      return np.array([])
+    if self.max_length is not None:
+      if (len(string) > self.max_length):  # Validation
```

Comment:
I'd say here we should probably just fail with a ValueError instead of trying to recover